torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -115,6 +115,119 @@ bool isSimpleMap(Node *node) {
+    "aten::reciprocal_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::relu_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::remainder_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::round_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::rsqrt_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::sigmoid_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::sin_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::sinh_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::sqrt_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::sub_(Tensor(a!) self, Tensor other, *, Scalar alpha) -> Tensor(a!)",
+    "aten::tan_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::tanh_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::trunc_(Tensor(a!) self) -> Tensor(a!)",
+    "aten::add_(Tensor(a!) self, Scalar other, Scalar alpha) -> Tensor(a!)",
+    "aten::sub_(Tensor(a!) self, Scalar other, Scalar alpha) -> Tensor(a!)",
+    "aten::mul_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+    "aten::div_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+
+    "aten::eq_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::eq_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+    "aten::ne_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::ne_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+    "aten::ge_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::ge_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+    "aten::gt_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::gt_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+    "aten::le_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::le_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+    "aten::lt_(Tensor(a!) self, Tensor other) -> Tensor(a!)",
+    "aten::lt_(Tensor(a!) self, Scalar other) -> Tensor(a!)",
+  }};
+  if (!mutable_maps.find(node)) {
+    return false;
+  }
+  // Check that all non-tensor inputs are constant
+  for (Value * input : node->inputs()) {
+    if (input->type()->isSubtypeOf(DynamicType::get())) {
+      continue;
+    }
+    if (input->node()->kind() != prim::Constant) {
+      return false;
+    }
+  }
+  return true;
+}
+
+Value* deinplace(Value * old_output) {
+  Node * node = old_output->node();
+  WithInsertPoint insert_guard { node };
+  auto name = std::string(node->kind().toQualString());
+  JIT_ASSERT(name.at(name.size() - 1) == '_');
+  name.pop_back(); // Remove the underscore
+  Graph * graph = node->owningGraph();
+  Node * new_node = graph->insertNode(graph->create(Symbol::fromQualString(name),
+                                                    node->inputs()));
+  Value * output = new_node->output();
+  node->output()->replaceAllUsesWith(output);
+  node->destroy();
+  return output;
+}
+
+bool needsToMutate(Node *node, const AliasDb& aliasDb) {
+  // A node is safe to be de-inplaced if all uses of values
+  // that can alias its output (that are not its output) happen
+  // before this node.
+  Value * output = node->output();
```

Comment:
You'll also need to check that there is no use of a wildcard value after `node`. I have a PR soon to hide "wildcards" as an implementation detail to avoid making people remember this in the future 😛

Reply:
I'd really like us to relax the definition of a wildcard to simply "all outputs alias all inputs and outputs", because as it is right now is super weird. It's not like we'll be manufacturing tensors you don't even get as arguments out of thin air...

Anyway, how do I get a list of wildcard values?

Reply:
The main place that wildcards show up is because we don't yet track aliases "into" lists and tuples. So when we index into a list and retrieve a tensor, it's not enough to say that it may alias all the inputs/outputs of the node.

I think it's enough to getWildcardNodes() and check that they are all before `node`.